src/cabal-details.js

```diff
@@ -276,10 +276,14 @@ class CabalDetails extends EventEmitter {
   }
 
   /**
-   * @returns {string[]} a list of all the channels in this cabal. Does not return channels with 0 members.
-   */
-  getChannels () {
-    return Object.keys(this.channels).filter(ch => this.channels[ch].members.size > 0).sort()
+   * @param {object} [opts]
+   * @property {boolean} includeArchived - Determines whether to include archived channels or not. Defaults to false.
+   * * @returns {string[]} a list of all the channels in this cabal. Does not return channels with 0 members.
+   */
+  getChannels (opts) {
+    if (!opts) opts = { includeArchived: false }
```

Comment:
this would cause issues in the situation where opts is a non-object type. not sure how careful you want to be in enforcing types. i would personally err on the side of safety if this is a public interface for the client.